src/Invoker.php

```diff
@@ -30,19 +31,37 @@ class Invoker
     private $errorLogFunc;
 
     /**
-     * @param callable $target      The callable to be invoked
-     * @param string $signatureType The signature type of the target callable,
-     *                              either "event" or "http".
+     * @param array|string $target    The callable to be invoked
+     * @param $signatureType          The signature type of the target callable,
+     *                                either "cloudevent" or "http". If null, 
+     *                                env var `FUNCTION_SIGNATURE_TYPE` is used.
      */
-    public function __construct(callable $target, string $signatureType)
+    public function __construct($target, ?string $signatureType = null)
     {
+        $signatureType = getenv('FUNCTION_SIGNATURE_TYPE', true) ?: 'http';
```

Comment:
@bshaffer I wasn't sure how to test the env var vs. declarative function precedence if I kept this in `router.php`. I couldn't figure out how to pass a CloudEvent request to the unit tests which use `require router.php` to execute the router.

Reply:
If you're trying to write an E2E test to ensure the environment variable `FUNCTION_SIGNATURE_TYPE` is ignored if a declarative type is used, take a look at [tests/routerTest.php](https://github.com/GoogleCloudPlatform/functions-framework-php/blob/master/tests/routerTest.php) and the files in [tests/fixtures](https://github.com/GoogleCloudPlatform/functions-framework-php/tree/master/tests/fixtures). You'll want to write a new entrypoint PHP file that follows that pattern.

Reply:
Those tests work with a blank HTTP requests though, right? I need to send a CloudEvent payload, otherwise when `require router.php` is called, the framework will error because it can't unmarshal the payload into a CloudEvent.

There didn't seem to be a way to pass a CloudEvent request to `router.php`.

Reply:
If your test injects the parameter directly, and insures it gets ignored, this is mocking the same behavior as what happens in `router.php`, where the parameter is fetched from the environment variable. I think this is a more precise way to test the behavior we want.

```php
    public function testCloudEventInvokerDeclarative(): void
    {
        InvokerTest::$cloudeventResponse = "bye";
        FunctionsFramework::cloudEvent('helloCloudEvent', function (CloudEventInterface $cloudevent) {
            InvokerTest::$cloudeventResponse = "Hello CloudEvent!";
        });
        // set the invoker to use "http" as the function type, and ensure that this is ignored
        $invoker = new Invoker('helloCloudEvent', 'http');
        $request = new ServerRequest(
            'POST',
            '',
            [],
            '{"eventId":"foo","eventType":"bar","resource":"baz"}'
        );
        $response = $invoker->handle($request);
        $this->assertSame('Hello CloudEvent!', InvokerTest::$cloudeventResponse);
    }
```

Reply:
Sounds good, done!